### PR TITLE
fix(sdk): fix organization/saml-connection updating logic.

### PR DESCRIPTION
**Pull request by jackworkgit**
1. Fix the SAML connection logic when updating the SAML connection.
2. Fix the organization login type when updating the status.

### CODE REVIEW

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java [hunk 1]
```diff
@@ -120,69 +120,89 @@ public Organization updateStatus(String orgId, OrgStatusEnum status, String upda
...
       log.error("update status of organization.error", e);
       throw VortexException.badRequest("Update status of organizations.error" + e.getMessage());
     }
   }
 
-  private void processOrgRelatedConnection(
+  private Organization processOrgRelatedConnection(
       String orgId,
       OrgStatusEnum status,
       String oldStatus,
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java:151]: I recommend change the oldStatus's type from `String` to `OrgStatusEnum`

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java [hunk 2]
```diff
@@ -120,69 +120,89 @@ public Organization updateStatus(String orgId, OrgStatusEnum status, String upda
...
       OrganizationsEntity organizationsEntity,
-      Organization oldOrg)
+      Organization newOrg)
       throws Auth0Exception {
+    Organization updateOrgLoginType = new Organization();
+    Map<String, Object> metadata =
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java:156]: Looks like `metadata` will never be `null`
if it's true, you can simple the logic.

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java [hunk 3]
```diff
@@ -120,69 +120,89 @@ public Organization updateStatus(String orgId, OrgStatusEnum status, String upda
...
+        newOrg.getMetadata() == null ? new HashMap<>() : newOrg.getMetadata();
+
     // active -> inactive: release the bound connection
     if (oldStatus.equals(OrgStatusEnum.ACTIVE.name()) && status == OrgStatusEnum.INACTIVE) {
       EnabledConnectionsPage enabledConnectionsPage =
           organizationsEntity.getConnections(orgId, null).execute().getBody();
 
       if (Objects.isNull(enabledConnectionsPage)
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java:164]: Maybe `enabledConnectionsPage` will never be `null` , pls double check the auth0-java-sdk,  if it's true, can delete `Objects.isNull`

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java [hunk 4]
```diff
@@ -399,17 +419,20 @@ public OrganizationConnection createConnection(
 
   public OrganizationConnection updateConnection(
       String orgId, UpdateConnectionDto request, String requestedBy) {
+    String strategy;
     try {
       log.info("Updating connection:orgId:{}, {},requestedBy:{}", orgId, request, requestedBy);
       Connection connection =
           auth0Client.getMgmtClient().connections().get(request.getId(), null).execute().getBody();
+      strategy = connection.getStrategy();
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java:427]: connection maybe Null here

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java (hunk 3 above)
Comment (jackworkgit) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java:164]: Now one organization doesn't have a connection by default. After creating a new organization we can change the status to inactive or recover it.

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java (hunk 4 above)
Comment (jackworkgit) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java:427]: yes, i will fix it.

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java (hunk 2 above)
Comment (jackworkgit) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java:156]: ok

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java (hunk 1 above)
Comment (jackworkgit) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java:151]: ok

**Review by pori — APPROVED** [review]

**Pull request merged**